finetune.py

```diff
@@ -216,8 +218,8 @@ def train_one_epoch(
         percent_complete = num_samples / samples_per_epoch * 100
 
         state, (loss, logits), rng = p_train_step(state, images, labels, rng)
-        loss = jax.tree_map(lambda x: x[0], loss)
-
+        state = jax.block_until_ready(state)  # wait for async computation to complete
```

Comment:
In other work I found a large discrepancy between reported duration with/without the synchonization (which incurs very little overhead) -- e.g. something that took >2s was reported as 20ms.

Prefer this for accurate timing

Reply:
Great catch. I totally forgot about this.